src/init/application/application.ts

```diff
@@ -75,6 +75,8 @@ export function createApplicationGenerator(templateName: string):
       ];
 
       this.props = await this.prompt(prompts);
+      this.props.elementClassName = this.props.elementName.replace(
+          /(^|\W)+(.)/g, (_match, _p0, p1) => p1.toUpperCase());
```

Comment:
+1, always up for some regex golf. Fixed that up a bit and landed on: `/(^|-)([a-z])/g`